lib/jekyll/filters/url_filters.rb

```diff
@@ -20,13 +20,17 @@ def absolute_url(input)
         ).normalize.to_s
       end
 
-      # Produces a URL relative to the domain root based on site.baseurl.
+      # Produces a URL relative to the domain root based on site.baseurl
+      # unless it is already an absolute url with an authority (host).
       #
       # input - the URL to make relative to the domain root
       #
       # Returns a URL relative to the domain root as a String.
       def relative_url(input)
         return if input.nil?
+        uri = Addressable::URI.parse(input.to_s)
```

Comment:
we are now calling `Addressable::URI.parse()` twice.. consider refactoring to keep it DRY?

Reply:
The second one parses the input with baseurl prepended and you need to know if the original input has a host before that. So this can't really be optimized.

Reply:
Only the `normalize.to_s` could be refactored, but I'm not sure this is better:

```ruby
uri = Addressable::URI.parse(input.to_s)
unless uri.host
  parts = [sanitized_baseurl, input]
  uri = Addressable::URI.parse(
    parts.compact.map { |part| ensure_leading_slash(part.to_s) }.join
  )
end
uri.normalize.to_s
```

It avoids duplication but there isn't much duplicated in the first place and I really like to return early for special cases.

Reply:
In the `absolute_url` filter, we have a one-line check to see if the URL is already absolute.

Let’s replace these two lines with that one.

Reply:
An absolute URI according to RFC 3986 includes a scheme. This is also what `Addressable::URI.absolute?` checks.
This would URIs like `//example.com/` slip through  - this is technically a URI with host `example.com` and empty path (`/`). 

It might be arguable if these should rather be interpreted as an (absolute) path beginning with two path separators.

But protocol-relative URIs are also valid and even with https-only becoming more and more standard they might still prove useful in some situations.

But you're certainly right that this should be handled similarly in both methods.